code/modules/spells/spell_types/mind_transfer.dm

```diff
@@ -1,98 +1,126 @@
-/obj/effect/proc_holder/spell/targeted/mind_transfer
+/obj/effect/proc_holder/spell/pointed/mind_transfer
 	name = "Mind Transfer"
-	desc = "This spell allows the user to switch bodies with a target."
-
+	desc = "This spell allows the user to switch bodies with a target next to him."
 	school = "transmutation"
 	charge_max = 600
 	clothes_req = FALSE
 	invocation = "GIN'YU CAPAN"
 	invocation_type = "whisper"
 	range = 1
 	cooldown_min = 200 //100 deciseconds reduction per rank
-	var/unconscious_amount_caster = 400 //how much the caster is stunned for after the spell
-	var/unconscious_amount_victim = 400 //how much the victim is stunned for after the spell
-
+	ranged_mousepointer = 'icons/effects/mindswap_target.dmi'
 	action_icon_state = "mindswap"
+	active_msg = "You prepare to swap minds with a target..."
+	/// For how long is the caster stunned for after the spell
+	var/unconscious_amount_caster = 40 SECONDS
+	/// For how long is the victim stunned for after the spell
+	var/unconscious_amount_victim = 40 SECONDS
 
-/*
-Urist: I don't feel like figuring out how you store object spells so I'm leaving this for you to do.
-Make sure spells that are removed from spell_list are actually removed and deleted when mind transferring.
-Also, you never added distance checking after target is selected. I've went ahead and did that.
-*/
-/obj/effect/proc_holder/spell/targeted/mind_transfer/cast(list/targets, mob/living/user = usr, distanceoverride, silent = FALSE)
+/obj/effect/proc_holder/spell/pointed/mind_transfer/cast(list/targets, mob/living/user, distance_override = FALSE, silent = FALSE)
 	if(!targets.len)
 		if(!silent)
 			to_chat(user, "<span class='warning'>No mind found!</span>")
-		return
-
+		return FALSE
 	if(targets.len > 1)
 		if(!silent)
 			to_chat(user, "<span class='warning'>Too many minds! You're not a hive damnit!</span>")
-		return
+		return FALSE
 
 	var/mob/living/target = targets[1]
+	if(!swap_check(user, target, distance_override, silent))
+		return FALSE
+	if(istype(target, /mob/living/simple_animal/hostile/guardian))
+		var/mob/living/simple_animal/hostile/guardian/stand = target
+		if(stand.summoner)
+			target = stand.summoner
+
+	var/mob/living/victim = target //The target of the spell whos body will be transferred to.

```

Comment:
`target` is already type cast as `/mob/living`, no need to have two variables.
You should do an `isliving()` check, however.

Reply:
Good catch, fixed.